.github/workflows/integrationTest.yml

```diff
@@ -638,52 +648,51 @@ jobs:
       - name: Get git date
         id: sha_date
         run: echo "::set-output name=sha_date::$(git show -s --format=%ct ${{ steps.sha.outputs.sha_short }} )"
-
       - name: Check env
-        run: echo "SHA ${GITHUB_SHA} | Date ${{ steps.sha_date.outputs.sha_date }}"
-
+        run: echo "SHA ${{ steps.sha.outputs.sha_short }} | Date ${{ steps.sha_date.outputs.sha_date }} "
+      - name: Verify Terraform version
+        run: terraform --version
       - name: Terraform apply
         if: steps.performance-tracking.outputs.cache-hit != 'true'
         uses: nick-invision/retry@v2
         with:
           max_attempts: 1
-          timeout_minutes: 50
+          timeout_minutes: 30
           retry_wait_seconds: 5
           command: |
             cd integration/terraform/ec2/linux
             terraform init
             if terraform apply --auto-approve \
               -var="ssh_key=${PRIVATE_KEY}" -var="github_repo=${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}.git" \
-              -var="github_sha=${GITHUB_SHA}" -var="install_agent=rpm -U ./amazon-cloudwatch-agent.rpm" \
-              -var="user=ec2-user" \
-              -var="ami=cloudwatch-agent-integration-test-al2*" \
-              -var="ca_cert_path=/etc/ssl/certs/ca-bundle.crt" \
-              -var="arc=amd64" \
-              -var="binary_name=amazon-cloudwatch-agent.rpm" \
+              -var="github_sha=${GITHUB_SHA}" -var="install_agent=${{ matrix.arrays.installAgentCommand }}" \
+              -var="user=${{ matrix.arrays.username }}" \
+              -var="ami=${{ matrix.arrays.ami }}" \
+              -var="ca_cert_path=${{ matrix.arrays.caCertPath }}" \
+              -var="arc=${{ matrix.arrays.arc }}" \
+              -var="binary_name=${{ matrix.arrays.binaryName }}" \
               -var="local_stack_host_name=${{ needs.StartLocalStack.outputs.local_stack_host_name }}" \
               -var="s3_bucket=${S3_INTEGRATION_BUCKET}" \
               -var="key_name=${KEY_NAME}" \
-              -var="test_name=cw-integ-test-al2" \
               -var="github_sha_date=${{ steps.sha_date.outputs.sha_date }}" \
-              -var="test_dir=./integration/test/performancetest" ; then terraform destroy -auto-approve
-
+              -var="test_name=cw-integ-test-${{ matrix.arrays.os }}" \
+              -var="performance_number_of_logs=${{ matrix.arrays.performance_number_of_logs}}"\
+              -var="test_dir=${{ matrix.arrays.test_dir }}" ; then terraform destroy -auto-approve
             else
               terraform destroy -auto-approve && exit 1
             fi
 
       #This is here just in case workflow cancel
       - name: Terraform destroy
-        if: ${{ cancelled() && steps.performance-tracking.outputs.cache-hit != 'true' }}
+        if: ${{ cancelled() && steps.ec2-linux-integration-test.outputs.cache-hit != 'true' }}
         uses: nick-invision/retry@v2
         with:
-          max_attempts: 3
+          max_attempts: 1
```

Comment:
If we're only allowing one attempt, why even have the retry? Why are we reducing this to 1? This is also for just the terraform destroy, so I don't see the issue.